### ISSUE TITLE
Support restoring main wallet from seed

### ISSUE THREAD

**Issue by woodser** [haveno-dex/haveno#752]
This issue requests supporting restoring the main wallet from seed in Account > Wallet Seed.

Currently the seed input is not validated correctly and does not support replacing the main wallet.

**Comment by woodser** [comment]
In order for this view to work correctly using date pickers, we need to translate between the date and the wallet's restore height.

That functionality is not currently in Monero RPC, leaving us with 3 options:

1. Wait for it to be supported in Monero RPC (see related PR https://github.com/monero-project/monero/pull/8088).
2. Switch to native bindings instead of RPC.
3. Change the inputs to a restore height instead of a date.

**Comment by xixipiece** [comment]
> In order for this view to work correctly using date pickers, we need to translate between the date and the wallet's restore height.

How does Feather Wallet do this?

If that's not available, how about this?

1. Hard-code a date -> height relationship for every Nth block in the past based on the current blockchain, except for the latest M blocks.
2. Use these anchors to estimate the height at a certain date with a conservative margin.
3. Update the anchors with every release.

If none of that is feasible and this is the only blocker to being able to restore from a seed, please consider exposing the restoration only with height. It's an advanced feature, whoever needs this either already knows their restore height or they can estimate it with the help of a block explorer.

Not being able to restore stops you from using deterministically generated seeds. Having to handle a new randomly generated secret every time you create a Haveno account is very bad. It's easier to lose that secret and maintaining backups is an overhead.

**Comment by xixipiece** [comment]
Sorry about bumping this, but this feature is really important to me and I'd appreciate if you could look at what I wrote above.

**Comment by woodser** [comment]
Until this is supported, a workaround is to import your seed phrase into the official monero GUI to create the wallet files, and then copy the wallet files into your application's xmr_mainnet/wallet folder.

**Comment by xixipiece** [comment]
Good idea. Are you sure that this won't result in a  corrupted profile?

Should I copy the wallet files to xmr_mainnet/wallet before or after Haveno generates the profile?

**Comment by woodser** [comment]
It's best to place the mainnet wallets before using Haveno, so nothing can reference a replaced wallet.
